### ISSUE TITLE
[2.1] Crash upon shift-clicking items in Refined Storage Disk Drive

### ISSUE THREAD

**Issue by SPMoo66** [sidben/VisibleArmorSlots#11]
In the Refined Storage Disk Drive, in which you can see the armor slots appearing correctly, there is a button for "Priority". Pressing this button reveals a second GUI where the slots appear again, but misplaced this time. 
http://i.imgur.com/xPhR0nD.png

Wearing any armor, if I try to shift-click any of it off, the game will crash, producing this log -
http://pastebin.com/pSdztEyf

Clicking normally on the items works, but shift-clicking will crash the game. This GUI technically isn't even an inventory, so the armor slots probably should not appear here to begin with.

I see that the mod is now compatible with any container, but most containers have this strange offset between the armor/icons and the grid that is supposed to be directly behind them. They also appear in poor locations at times.

Some "inventories" where the armor slots appear in a bad place -
http://i.imgur.com/IaLcQ42.png
The RFTools Control Programmer

Some blocks with UIs where the armor slots shouldn't be appearing at all -

http://i.imgur.com/XyIZvfE.png
The RFTools Timer, Sequencer, Logic Gate, and more.

http://i.imgur.com/OLdI3wa.png
EnderIO Capacitor Banks (doubled)

And even beyond that, the Refined Storage Grid has the armor slot GUI appear, but the slots are not actually accessible.
http://i.imgur.com/YE98k3g.png
Despite appearing, the bottom left inventory slots for armor cannot be used, and the armor currently equipped cannot be seen.

I really do appreciate the appearance of the armor slots in inventories like iron chests, so I'm hoping the few remaining issues can be ironed out. Thanks!

**Comment by sidben** [comment]
Thanks for the details, I'm almost done with the new version that should fix all position/alignment issues.

I haven't considered UI's that don't have slots, I'll add that to my to-do list before release.

**Comment by SPMoo66** [comment]
Is there any chance of a backport of the fixes to 1.10?

**Comment by sidben** [comment]
Yes, a 1.10 backport of the latest version is already on the way. I finished most of the code changes, now I need to manually test with some modpacks and fix any issues I find.
